lib/Transformer.js

```diff
@@ -35,8 +35,8 @@ module.exports = {
                         }
 
                         // in case in the dialog there more paricipants than in the conversation
-                        let participantsPId = conversationDetails.participantsPId[p.role][roles[p.role]];
-                        p.id = participantsPId ? participantsPId : p.id;
+                        const participantsProp = Object.prototype.hasOwnProperty.call(conversationDetails.participantsPId, p.role);
+                        p.id = participantsProp ? conversationDetails.participantsPId[p.role][roles[p.role]] : p.id;
```

Comment:
Can you add a test for the previously failing case?

Reply:
This code is very difficult to understand as written perhaps try this:
if (conversationDetails.participantsPId.hasOwnProperty(p.role)) {
    p.id = conversationDetails.participantsPId[p.role][roles[p.role]];
}